### ISSUE TITLE
Improvements to Vaadin Details accessibility

### ISSUE THREAD

**Issue by juuso-vaadin** [vaadin/vaadin-details#38]
The component should define a connection between summary (element used to trigger the details panel) and the contents of the panel. This is done with **aria-controls** tag.

Contents element should have **role="region"** and an **id** that is used in **aria-controls** tag.

Another aria-tags that could be utilized are
- **aria-hidden** for hidden contents
- **aria-expanded** for element controlling the panel (summary)
- **aria-labelledby** for content and **id** for summary

Example: https://www.w3.org/TR/wai-aria-practices-1.1/examples/accordion/accordion.html

**Comment by anezthes** [comment]
I'd recommend using `<section>` instead of `role="region"`.

**Comment by web-padawan** [comment]
Once we update the `vaadin-details` component, we should take care of `vaadin-accordion` as well.

See https://daverupert.com/2019/12/why-details-is-not-an-accordion/
